src/axelrod_dojo/algorithms/genetic_algorithm.py

```diff
@@ -77,22 +84,57 @@ def crossover(population, num_variants):
 
     def evolve(self):
         self.generation += 1
-        print("Scoring Generation {}".format(self.generation))
+        if self.print_output:
+            print("Scoring Generation {}".format(self.generation))
 
         # Score population
         scores = self.score_all()
         results = list(zip(scores, range(len(scores))))
         results.sort(key=itemgetter(0), reverse=True)
 
-        # Report
-        print("Generation", self.generation, "| Best Score:", results[0][0],
-              repr(self.population[results[0][1]]))
         # Write the data
-        row = [self.generation, mean(scores), pstdev(scores), results[0][0],
-               repr(self.population[results[0][1]])]
-        self.outputer.write(row)
-
-        ## Next Population
+        # Note: if using this for analysis, for reproducability it may be useful to alter the data passed back
+
+        opponent_strategy_name = 'unknown'
+
+        for strategy, init_kwargs in self.opponents_information:
+            opponent = strategy(**init_kwargs)
+            opponent_strategy_name = type(opponent).name
+            try:
+                player_seed = init_kwargs['seed']
+            except KeyError:
+                player_seed = None
+
+        # Row Data Cols:
```

Comment:
This risk becoming lies when/if something gets added, I think what is in each column is pretty explicit so let's just remove the inline comments.

Reply:
A better approach would be to create a temporary variable for some of the columns (the ones that need it perhaps):

```
population_score_range = results[0][0] - results[len(results) - 1][0]
```

and then put that in `row`.